pkg/sdkserver/sdkserver.go

```diff
@@ -1238,13 +1239,14 @@ func (s *SDKServer) updateList(ctx context.Context) error {
 
 // Returns a new string list with the string keys in toDeleteValues removed from valuesList.
 func deleteValues(valuesList []string, toDeleteValues map[string]bool) []string {
-	newList := make([]string, 0, len(valuesList)-len(toDeleteValues))
-	for i, val := range valuesList {
-		if _, ok := toDeleteValues[val]; !ok {
-			newList = append(newList, valuesList[i])
+	newValuesList := make([]string, 0, len(valuesList))
```

Comment:
Possibly easier suggestion for defensive programming.

```suggestion
  newLen = len(valuesList)-len(toDeleteValues)
  if newLen < 0 {
    newLen = 0
  }

	newList := make([]string, 0, newLen)
```

And then we continue on as before.

Or I think as you said, make `newValueList := []string{}` and let it expand naturally - which is a bit more readable IMHO, but I don't have strong opinions about either approach.

Reply:
Thanks for your suggestions!  Already fixed, please review